app/services/foreman/telemetry_helper.rb

```diff
@@ -24,7 +24,7 @@ def telemetry_duration_histogram(name, scale = 1000, tags = {}, results = nil)
       scale = 1 / 60
     end
     before = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-    yield
+    yield if block_given?
```

Comment:
Can you throw exception if block is not given instead?

Then in the `ldap_instrumentation.rb` use method `telemetry_observe_histogram` instead of this block helper. It was a typo, I mean the other one.

Reply:
I changed the method call, it raises exception itself if no block is passed, it's rescued below and logged